ExZos/soen487-appointment-system
# Frontend Authentication

React components and utilities for authentication + some backend changes to authentication-related endpoints
<br />

<ins>What to test:</ins>

1. Run frontend and go through the login process for both admin and client
Once logged in check (F12) console log and make sure printed user values match with DB
2. Check that client login can be done through Postman 
(SSO process is slightly different with browser vs. Postman/cURL)
<br />

<ins>Frontend auth implementation notes</ins>
- Authentication is handled in `index.js`
- Frontend paths that are for admins use the `AdminRoute` component
- AND those that are for clients use the `ClientRoute` component
- Depending on which of those routes are used, the component can have access to slightly different user objects
- Components that are routed by `AdminRoute` have access to `username` and `token`
- WHILE those routed by `ClientRoute` have access to `email` and `token`
- These are accessed like this: `props.user.username` `props.user.email` `props.user.token`
- ALSO make sure to add the `props` param to the component
- Check `home.js` if you need an example of getting these values
